crates/shared/src/event_handling.rs

```diff
@@ -286,19 +290,26 @@ where
 
     async fn update_events_from_latest_blocks(
         &mut self,
-        blocks: &[BlockNumberHash],
+        latest_blocks: &[BlockNumberHash],
         is_reorg: bool,
     ) -> Result<()> {
-        let (blocks, events) = self.past_events_by_block_hashes(blocks).await;
-        track_block_range(&format!("range_{}", blocks.len()));
-        tracing::debug!(
-            "final blocks for updating: {:?} - {:?}",
-            blocks.first(),
-            blocks.last()
+        anyhow::ensure!(
+            !latest_blocks.is_empty(),
+            "entered update events with empty block list"
         );
-        if blocks.is_empty() {
-            return Ok(());
+        let (blocks, events) = self.past_events_by_block_hashes(latest_blocks).await;
+        track_block_range(&format!("range_{}", blocks.len()));
+        if blocks != latest_blocks {
+            tracing::debug!(
+                "some of the blocks filtered out, partial update: {:?} - {:?}",
+                blocks.first(),
+                blocks.last()
+            );
+            if blocks.is_empty() {
+                return Ok(());
```

Comment:
Shouldn't this be an error since we didn't update completely?

Reply:
Yes, I just thought we are gonna drop the PR completely. Reverted.